src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,50 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  die() {
+    const index = Animal.alive.indexOf(this);
+
+    if (index !== -1) {
+      Animal.alive.splice(index, 1);
+    }
```

Comment:
The use of `indexOf` and `splice` within the `die` method can lead to inefficient performance when dealing with a large number of animals. Consider using `filter` for better performance and to avoid nested loops.

Reply:
Fix [this comment](https://github.com/mate-academy/js_herbivores_and_carnivores/pull/2928/files#r1667710853).
Better to use filter method instead splice and indexOf